docs/contracts/inheritance.rst

```diff
@@ -108,93 +115,170 @@ Details are given in the following example.
+        }
     }
 
-A call to ``Final.destroy()`` will call ``Base2.destroy`` because we specify it
+A call to ``Final.pause()`` will call ``Base2.pause`` because we specify it
 explicitly in the final override, but this function will bypass
-``Base1.destroy``. The way around this is to use ``super``:
+``Base1.pause``, resulting in the following sequence of events:
+``Paused_Final -> Paused_Base2 -> Paused``, instead of the expected sequence:
+``Paused_Final -> Paused_Base2 -> Paused_Base1 -> Paused``.
+The way around this is to use ``super``:
 
 .. code-block:: solidity
 
     // SPDX-License-Identifier: GPL-3.0
     pragma solidity >=0.7.0 <0.9.0;
-    // This will report a warning due to deprecated selfdestruct
 
-    contract owned {
-        constructor() { owner = payable(msg.sender); }
+    contract Owned {
         address payable owner;
+        constructor() { owner = payable(msg.sender); }
     }
 
-    contract Destructible is owned {
-        function destroy() virtual public {
-            if (msg.sender == owner) selfdestruct(owner);
+    contract Pausable is Owned {
+        bool private _paused;
+
+        event Paused();
+        event Unpaused();
+
+        constructor() { _paused = false; }
+
+        function paused() public view virtual returns (bool) {
+            return _paused;
+        }
+
+        function pause() virtual public {
+            if (msg.sender == owner && !_paused)
+            {
+                emit Paused();
+                _paused = true;
+            }
+        }
+
+        function unpause() virtual public {
+            if (msg.sender == owner && _paused)
+            {
+                emit Unpaused();
+                _paused = false;
+            }
         }
     }
 
-    contract Base1 is Destructible {
-        function destroy() public virtual override { /* do cleanup 1 */ super.destroy(); }
+    contract Base1 is Pausable {
+        event Paused_Base1();
+        function pause() public virtual override {
+            /* Here, we emit an event to simulate some Base1 pause logic */
+            emit Paused_Base1();
+            super.pause();
+        }
     }
 
 
-    contract Base2 is Destructible {
-        function destroy() public virtual override { /* do cleanup 2 */ super.destroy(); }
+    contract Base2 is Pausable {
+        event Paused_Base2();
+        function pause() public virtual override {
+            /* Here, we emit an event to simulate some Base2 pause logic */
+            emit Paused_Base2();
+            super.pause();
+        }
     }
 
     contract Final is Base1, Base2 {
-        function destroy() public override(Base1, Base2) { super.destroy(); }
+        event Paused_Final();
+        function pause() public override(Base1, Base2) {
+            /* Here, we emit an event to simulate some Final pause logic */
+            emit Paused_Final();
+            super.pause();
+        }
     }
 
 If ``Base2`` calls a function of ``super``, it does not simply
```

Comment:
Not strictly related to this PR, but shouldn't this be referring to `Final` instead of `Base2`?
```suggestion
If ``Final`` calls a function of ``super``, it does not simply
```

Reply:
Yes.